File: src/main/kotlin/br/com/jarvis/config/security/AuthorizationServerConfig.java
//package br.com.jarvis.config.security;
//
//import br.com.jarvis.service.auth.CustomUserDetailsService;
//import org.springframework.beans.factory.annotation.Value;
//import org.springframework.context.annotation.Bean;
//import org.springframework.context.annotation.Configuration;
//import org.springframework.security.authentication.AuthenticationManager;
//import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
//import org.springframework.security.oauth2.config.annotation.configurers.ClientDetailsServiceConfigurer;
//import org.springframework.security.oauth2.config.annotation.web.configuration.AuthorizationServerConfigurerAdapter;
//import org.springframework.security.oauth2.config.annotation.web.configuration.EnableAuthorizationServer;
//import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerEndpointsConfigurer;
//import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerSecurityConfigurer;
//import org.springframework.security.oauth2.provider.token.DefaultAccessTokenConverter;
//import org.springframework.security.oauth2.provider.token.TokenEnhancer;
//import org.springframework.security.oauth2.provider.token.TokenStore;
//
//@Configuration
//@EnableAuthorizationServer
//public class AuthorizationServerConfig extends AuthorizationServerConfigurerAdapter {
//
//	private final String clientId;
//
//	private final String clientSecret;
//
//	private static final String GRANT_TYPE_PASS = "password";
//	private static final String AUTHORIZATION_CODE = "authorization_code";
//	private static final String REFRESH_TOKEN = "refresh_token";
//	private static final String IMPLICIT = "implicit";
//	private static final String ADMIN = "admin";
//	private static final String APP = "app";
//	private static final String SITE = "site";
//	private static final int ACCESS_TOKEN_VALIDITY_SECONDS = 8 * 60 * 60;
//
//	private final TokenStore tokenStore;
//
//	private final AuthenticationManager authenticationManager;
//
//	private final BCryptPasswordEncoder bCryptPasswordEncoder;
//
//	private final CustomUserDetailsService customUserDetailsService;
//
//
//	public AuthorizationServerConfig(
//			@Value("${spring.security.oauth2.client.clientId}") String clientId,
//			@Value("${spring.security.oauth2.client.clientSecret}") String clientSecret,
//			TokenStore tokenStore,
//			AuthenticationManager authenticationManager,
//			BCryptPasswordEncoder bCryptPasswordEncoder,
//			CustomUserDetailsService customUserDetailsService) {
//		this.clientId = clientId;
//		this.clientSecret = clientSecret;
//		this.tokenStore = tokenStore;
//		this.authenticationManager = authenticationManager;
//		this.bCryptPasswordEncoder = bCryptPasswordEncoder;
//		this.customUserDetailsService = customUserDetailsService;
//	}
//
//	@Override
//	public void configure(ClientDetailsServiceConfigurer configurer) throws Exception {
//		configurer
//				.inMemory()
//				.withClient(clientId)
//				.secret(bCryptPasswordEncoder.encode(clientSecret))
//				.authorizedGrantTypes(AuthorizationServerConfig.GRANT_TYPE_PASS, AuthorizationServerConfig.AUTHORIZATION_CODE, AuthorizationServerConfig.REFRESH_TOKEN, AuthorizationServerConfig.IMPLICIT)
//				.scopes(AuthorizationServerConfig.ADMIN, AuthorizationServerConfig.APP, AuthorizationServerConfig.SITE)
//				.accessTokenValiditySeconds(AuthorizationServerConfig.ACCESS_TOKEN_VALIDITY_SECONDS)
//				.refreshTokenValiditySeconds(0);
//	}
//
//	@Bean
//	public DefaultAccessTokenConverter accessTokenConverter() {
//		return new DefaultAccessTokenConverter();
//	}
//
//	@Bean
//	public TokenEnhancer tokenEnhancer() {
//		return new CustomTokenEnhancer();
//	}
//
//
//	@Override
//	public void configure(AuthorizationServerEndpointsConfigurer endpoints) {
//		endpoints.tokenStore(tokenStore)
//				.tokenEnhancer(tokenEnhancer())
//				.accessTokenConverter(accessTokenConverter())
//				.authenticationManager(authenticationManager)
//				.userDetailsService(customUserDetailsService);
//	}
//
//	@Override
//	public void configure(AuthorizationServerSecurityConfigurer oauthServer) {
//		oauthServer.checkTokenAccess("permitAll()");
//	}
//
//}

File: src/main/kotlin/br/com/jarvis/config/security/AuthorizationServerConfig.kt
package br.com.jarvis.config.security


import br.com.jarvis.service.auth.CustomUserDetailsService
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.oauth2.config.annotation.configurers.ClientDetailsServiceConfigurer
import org.springframework.security.oauth2.config.annotation.web.configuration.AuthorizationServerConfigurerAdapter
import org.springframework.security.oauth2.config.annotation.web.configuration.EnableAuthorizationServer
import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerEndpointsConfigurer
import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerSecurityConfigurer
import org.springframework.security.oauth2.provider.token.DefaultAccessTokenConverter
import org.springframework.security.oauth2.provider.token.DefaultTokenServices
import org.springframework.security.oauth2.provider.token.TokenEnhancer
import org.springframework.security.oauth2.provider.token.TokenStore
import org.springframework.security.oauth2.provider.token.store.InMemoryTokenStore


@Configuration
@EnableAuthorizationServer
open class AuthorizationServerConfig(
    private val authenticationManager: AuthenticationManager,
    private val userDetailsService: CustomUserDetailsService,
) : AuthorizationServerConfigurerAdapter() {

    private val GRANT_TYPE_PASS = "password"
    private val AUTHORIZATION_CODE = "authorization_code"
    private val REFRESH_TOKEN = "refresh_token"
    private val IMPLICIT = "implicit"
    private val ADMIN = "admin"
    private val APP = "app"
    private val SITE = "site"
    private val ACCESS_TOKEN_VALIDITY_SECONDS = 8 * 60 * 60

    override fun configure(clients: ClientDetailsServiceConfigurer) {
        clients.inMemory()
            .withClient("jarvis-dev") //mudei aqui 524257436231028
            .secret("x*Wg9`xZ\\e@\\^Hd,)rezCW(a=Nfk3qxF") // mudei  aqui o clientsecret a45f53d2c7b335af811797e77d50cbfb
//            .authorizedGrantTypes(GRANT_TYPE_PASS, AUTHORIZATION_CODE, REFRESH_TOKEN, IMPLICIT)
////            .scopes( APP)// SE EU REMOVO ISSO AQUI TODAS AS REQUESTS FUNCIONAM
//            .accessTokenValiditySeconds(ACCESS_TOKEN_VALIDITY_SECONDS)
////            .authorizedGrantTypes("authorization_code", "refresh_token")
////            .scopes("read", "write")
//            .accessTokenValiditySeconds(3600) // Token validity time in seconds
    }
//
//    override fun configure(endpoints: AuthorizationServerEndpointsConfigurer) {
//        endpoints.tokenStore(tokenStore())
//    }

    // preciso comentar isso
    @Bean
    @Primary
    open fun defaultTokenServices(): DefaultTokenServices {
        val tokenServices = DefaultTokenServices()
        tokenServices.setTokenStore(tokenStore()) // Set your token store implementation
        return tokenServices
    }

    @Bean
    open fun tokenStore(): TokenStore {
        return InMemoryTokenStore()
    }

    @Bean
    open fun accessTokenConverter(): DefaultAccessTokenConverter? {
        return DefaultAccessTokenConverter()
    }

    @Bean
    open fun tokenEnhancer(): TokenEnhancer? {
        return YourCustomTokenEnhancer()
    }


    override fun configure(endpoints: AuthorizationServerEndpointsConfigurer) {
        endpoints.tokenStore(tokenStore())
            .tokenEnhancer(tokenEnhancer())
            .accessTokenConverter(accessTokenConverter())
            .authenticationManager(authenticationManager)
            .userDetailsService(userDetailsService)
    }

    override fun configure(oauthServer: AuthorizationServerSecurityConfigurer) {
        oauthServer.checkTokenAccess("permitAll()")
    }

}